packages/e2e-tests/src/specs/editor/fontCheck/fontCheckMetrics.js

```diff
@@ -0,0 +1,205 @@
+  await page.click('[aria-label="Web Stories Editor"]');
+
+  const isMacOS = await isPlatformMacOS();
+
+  await page.keyboard.down(isMacOS ? 'Meta' : 'Control');
+  await page.keyboard.down('Shift');
+  await page.keyboard.down('Alt');
+  await page.keyboard.down('J');
+  await page.keyboard.up('J');
+  await page.keyboard.up('Alt');
+  await page.keyboard.up('Shift');
+  await page.keyboard.down(isMacOS ? 'Meta' : 'Control');
+
+  // Dev Tools were not open before, but now they should be open.
+  if (!areDevToolsOpen) {
+    await page.waitForSelector('#web-stories-editor textarea');
+  }
+}
+
+async function getCurrentStoryData() {
+  await toggleDevTools();
+
+  const textareaContent = await page.evaluate(
+    () => document.querySelector('#web-stories-editor textarea').value
+  );
+
+  await toggleDevTools();
+
+  return JSON.parse(textareaContent);
+}
+
+describe('Font Check Metrics', () => {
+  let stopRequestInterception;
+  let mockResponse;
+
+  beforeAll(async () => {
+    await page.setRequestInterception(true);
+    stopRequestInterception = addRequestInterception((request) => {
+      if (request.url().includes('web-stories/v1/fonts') && mockResponse) {
+        request.respond(mockResponse);
+        return;
+      }
+
+      request.continue();
+    });
+  });
+
+  afterEach(() => {
+    mockResponse = undefined;
+  });
+
+  afterAll(() => {
+    stopRequestInterception();
+  });
+
+  it('should receive updated font metrics and not alter history', async () => {
+    const storyTitle = 'Font Check Metrics';
+    await addStoryWithFont(storyTitle);
+
+    const storyData = await getCurrentStoryData();
+
+    const fontBefore = storyData.pages[0].elements[1].font;
+    expect(fontBefore.family).toBe('Rock Salt');
+    expect(fontBefore.fallbacks).toIncludeAllMembers(['cursive']);
+    expect(fontBefore.metrics.upm).toBe(1024);
+
+    mockResponse = {
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify([
+        {
+          family: 'Rock Salt',
+          fallbacks: ['sans-serif'], // Original is "cursive"
+          weights: [900],
+          styles: ['italic'],
+          variants: [[0, 400]],
+          service: 'fonts.google.com',
+          metrics: {
+            upm: 200, // Original is 1024
+            asc: 1623,
+            des: -788,
+            tAsc: 824,
+            tDes: -240,
+            tLGap: 63,
+            wAsc: 1623,
+            wDes: 788,
+            xH: 833,
+            capH: 1154,
+            yMin: -787,
+            yMax: 1623,
+            hAsc: 1623,
+            hDes: -788,
+            lGap: 32,
+          },
+        },
+      ]),
+    };
+
+    await Promise.all([
+      page.waitForResponse(
+        (response) =>
+          //eslint-disable-next-line jest/no-conditional-in-test -- False positive.
+          response.url().includes('web-stories/v1/fonts') &&
+          response.status() === 200
+      ),
+      page.reload(),
+    ]);
```

Comment:
Why promise all here? Should we wait until one is finish then then next?

Reply:
That would not work reliably, because our fonts request is done super early on mount, and `page.reload()` waits for `load`/`DOMContentLoaded`, so if we'd do `await page.reload(); await pageWaitForResponse()` then the request might have already finished at this point. So using `Promise.all()` avoids this race condition.